beater/beater.go

```diff
@@ -192,11 +194,27 @@ func initTracer(info beat.Info, config *Config, logger *logp.Logger) (*elasticap
 	if err != nil {
 		return nil, nil, err
 	}
-	if config.SelfInstrumentation.isEnabled() {
-		if config.SelfInstrumentation.Environment != nil {
-			tracer.Service.Environment = *config.SelfInstrumentation.Environment
+	// tracing disabled, setup complete
+	if !config.SelfInstrumentation.isEnabled() {
+		return tracer, nil, nil
+	}
+
+	if config.SelfInstrumentation.Environment != nil {
```

Comment:
@axw any reason to do this before returning early, when instrumentation is disabled?

Reply:
nope, not unless/until we make the "enabled" configuration dynamic